fix build with libc++ 19
As noted in the libc++ 19 release notes [1], std::char_traits<> is now
only provided for char, char8_t, char16_t, char32_t and wchar_t, and any
instantiation for other types will fail.
[1] https://libcxx.llvm.org/ReleaseNotes/19.html#deprecations-and-removals
Index: core/src/Utf.cpp
--- core/src/Utf.cpp.orig
+++ core/src/Utf.cpp
@@ -13,6 +13,7 @@
 #include <iomanip>
 #include <cstdint>
 #include <sstream>
+#include <vector>
 
 namespace ZXing {
 
@@ -20,7 +21,7 @@ namespace ZXing {
 #if __cplusplus <= 201703L
 using char8_t = uint8_t;
 #endif
-using utf8_t = std::basic_string_view<char8_t>;
+using utf8_t = std::vector<char8_t>;
 
 using state_t = uint8_t;
 constexpr state_t kAccepted = 0;
@@ -118,7 +119,8 @@ static void AppendFromUtf8(utf8_t utf8, std::wstring& 
 std::wstring FromUtf8(std::string_view utf8)
 {
 	std::wstring str;
-	AppendFromUtf8({reinterpret_cast<const char8_t*>(utf8.data()), utf8.size()}, str);
+	const char8_t* data = reinterpret_cast<const char8_t*>(utf8.data());
+	AppendFromUtf8({data, data + utf8.size()}, str);
 	return str;
 }
 
